cpu/stm32/periph/eth.c

```diff
@@ -46,40 +48,43 @@
 #define CLOCK_RANGE ETH_MACMIIAR_CR_Div102
 #endif /* CLOCK_CORECLOCK < (20000000U) */
 
-/* Internal flags for the DMA descriptors */
-#define DESC_OWN           (0x80000000)
-#define RX_DESC_FL         (0x3FFF0000)
-#define RX_DESC_FS         (0x00000200)
-#define RX_DESC_LS         (0x00000100)
-#define RX_DESC_RCH        (0x00004000)
-#define TX_DESC_TCH        (0x00100000)
-#define TX_DESC_IC         (0x40000000)
-#define TX_DESC_CIC        (0x00C00000)
-#define TX_DESC_LS         (0x20000000)
-#define TX_DESC_FS         (0x10000000)
-
-struct eth_dma_desc {
-    uint32_t status;
-    uint32_t control;
-    char *buffer_addr;
-    struct eth_dma_desc *desc_next;
-    uint32_t reserved1_ext;
-    uint32_t reserved2;
-    uint32_t ts_low;
-    uint32_t ts_high;
-} __attribute__((packed));
-
-typedef struct eth_dma_desc edma_desc_t;
+/* Default DMA buffer setup */
+#ifndef ETH_RX_DESCRIPTOR_COUNT
+#define ETH_RX_DESCRIPTOR_COUNT (6U)
+#endif
+#ifndef ETH_TX_DESCRIPTOR_COUNT
+#define ETH_TX_DESCRIPTOR_COUNT (8U)
+#endif
+#ifndef ETH_RX_BUFFER_SIZE
+#define ETH_RX_BUFFER_SIZE  (256U)
+#endif
+
+#if (ETH_RX_BUFFER_SIZE % 16) != 0
+/* For compatibility with 128bit memory interfaces, the buffer size needs to
+ * be a multiple of 16 Byte. For 64 bit memory interfaces need the size to be
+ * a multiple of 8 Byte, for 32 bit a multiple of 4 byte is sufficient. */
+#warning "ETH_RX_BUFFER_SIZE is not a multiple of 16. (See comment above.)"
+#endif
+
+#if ETH_RX_DESCRIPTOR_COUNT * ETH_RX_BUFFER_SIZE < 1524U
+#warning "Total RX buffers lower than MTU, you won't receive huge frames!"
+#endif
+
+#define MINIMUM(a, b) (((a) <= (b)) ? (a) : (b))
```

Comment:
Why not `MIN()`?

Reply:
I was told that `#define MIN 60` (as in `HOUR`, `MIN`, `SEC`) is also popular and this avoids this ambiguity. But I have no idea why one would use minutes in an Ethernet driver. So if you prefer `MIN` over `MINIMUM`, I could change this.

Reply:
`git grep "define MIN("` shows that the `MIN()` macro is actually pretty common.

Reply:
Done

Reply:
IMO, we should give the `MIN()` macro in `kernel_defines.h` another spin. I think half of the maintainers blocking that PR back then are no longer active, so chances are good a PR would now make it.